src/test/py/peer/peer_test.py

```diff
@@ -148,6 +148,46 @@ def test_verify_integrity(self):
         self.ledger.queue[0] = (self.peer, b"corrupted_hash")
         self.assertFalse(self.ledger.verify_integrity())
 
+    def test_sign_artwork(self):
+        """
+        Test signing artwork with peer's private key.
+        """
+        canvas = Image.new(mode="RGB", size=(10, 20), color=(255, 255, 255))
+        pixels = canvas.load()
+
+        for x in range(10):
+            for y in range(20):
+                choice = bool(random.getrandbits(1))
+                if choice:
+                    pixels[x, y] = (0, 0, 255)
+
+        signature = self.peer.sign_artwork(canvas)
+        self.assertIsInstance(signature, bytes)
+
+    def test_verify_artwork(self):
+        """
+        Test verify_artwork
+        """
+        # generate image
+        canvas = Image.new(mode="RGB", size=(10, 20), color=(255, 255, 255))
+        pixels = canvas.load()
+
+        for x in range(10):
+            for y in range(20):
+                choice = bool(random.getrandbits(1))
+                if choice:
+                    pixels[x, y] = (0, 0, 255)
+
+        canvas.save("canvas.png", "PNG")
+
+        signature = self.peer.sign_artwork(canvas)
+        verification = self.peer.verify_artwork(
+            signature=signature,
+            public_key_string=self.peer.keys["public"],
+            artwork=canvas,
+        )
+        self.assertEqual(verification, True)
+
```

Comment:
One more test that would be good to have:
* create and sign an image with an app that generates a compatible signature
* load that image and sig
* check validity with our `verify`